W-21433121 agent network 2.0 content ie (#660)
mulesoft/docs-code-builder: agent-network/2.0/modules/ROOT/pages/af-publish-agent-network-assets.adoc

[[publish-assets]]
= Publish Agent Network Assets
:page-aliases: anypoint-code-builder::af-publish-agent-network-assets.adoc

> Build and publish your agent network project as Anypoint Exchange assets. When you publish, an asset is created for each broker, agent, and MCP server that's in your agent network. The asset ID is the value of the `name` key in `agent-network.yaml`.

Reviewer: ```suggestion
Build and publish your agent network project as Anypoint Exchange assets. When you publish, Exchange creates an asset for each broker, agent, and MCP server in your agent network. The asset ID is the value of the `name` key in `agent-network.yaml`.
```

> For example, suppose you have an agent name of `badging-agent` in `agent-network.yaml`. After it's published, the Exchange `assetId` is `badging-agent`.

Reviewer: ```suggestion
For example, suppose you have an agent named `badging-agent` in `agent-network.yaml`. After you publish, the Exchange `assetId` is `badging-agent`.
```

[[before-you-begin]]
== Before You Begin

Make sure you review the xref:af-get-started.adoc#before-you-begin[prerequisites].

> If you want to publish your project assets to a different business group, update the relevant `groupId` in `exchange.json`.

Reviewer: ```suggestion
To publish your project assets to a different business group, update the relevant `groupId` in `exchange.json`.
```

== Versioning Assets

> Do not specify a semantic version (for example, `1.0.1`) each time you publish your project. Instead, enter a version group (for example, `V1` or `V2`) that represents the high-level version of your agent network project. Exchange handles the semantic versioning and automatically increments the version when there are changes.

Reviewer: ```suggestion
Don't specify a semantic version (for example, `1.0.1`) each time you publish your project. Instead, enter a version group (for example, `V1` or `V2`) that represents the high-level version of your agent network project. Exchange handles the semantic versioning and automatically increments the version when there are changes.
```

> For existing projects, specify the same version group you were previously using. Specifying a different version group triggers a major semantic version upgrade (for example, `1.x.x` to `2.x.x`) for all associated assets.

Reviewer: ```suggestion
For existing projects, specify the same version group you were previously using. If you specify a different version group, then Exchange triggers a major semantic version upgrade (for example, `1.x.x` to `2.x.x`) for all associated assets.
```

[[publish-dev-agent]]
== Publish Your Network Using MuleSoft Vibes

Use MuleSoft Vibes to publish your agent network specification to Anypoint Exchange. For more information, see xref:anypoint-code-builder::api-ai-create-spec.adoc[].

. In the Anypoint Code Builder activity bar, click the agent icon image:af-acb-dev-agent-icon.png["",18,18].

> . Tell the agent that you want to publish your agent network. MuleSoft Vibes does the rest.

Reviewer: ```suggestion
. Tell MuleSoft Vibes that you want to publish your agent network. MuleSoft Vibes does the rest.
```

To get started, try one of these suggested messages.

* Publish my Employee Onboarding project.
* Publish my agent network project version 1.0.0.
* Publish the hello-world-network project to Exchange.
* Publish agent network project at /path/to/project with version 2.1.0.

[[publish-acb]]
== Publish Your Network Using Anypoint Code Builder

Use Anypoint Code Builder to publish your agent network assets directly from your project to Anypoint Exchange.

. If you're not logged in already, log in to your Anypoint Platform account.
. In your Anypoint Code Builder project, choose one of the following:
  . In Explorer, right-click a project file and select *Publish Agent Network Assets*.
  . In the Command Palette, run this command: *MuleSoft: Publish Agent Network Assets*.
. In *Publish Agent Network Project Assets*, enter a version group for the project and review the project metadata. If you want to publish project assets to a different business group, update the `groupId` in `exchange.json`.
. Select *Publish*.

> . Your agents, MCP servers, and LLM provider definitions are published to Exchange. An Exchange asset is created for each element in your agent network.

Reviewer: ```suggestion
. Anypoint Code Builder publishes your agents, MCP servers, and LLM provider definitions to Exchange and creates an asset for each element in your agent network.
```

Next, discover your agent network assets in *Agents & Tools* in Exchange to get a comprehensive view of your integration, including agent and MCP server assets. Exchange automatically tags brokers for easy identification.

[[publish-cli]]
== Publish a Network Using the Anypoint CLI

> If you run operations within a CI/CD environment, you can use Anypoint CLI's plugin to set up, create, build, publish, and deploy agent networks. For more information, see xref:af-build-agent-networks-in-a-ci-cd-environment.adoc[].

Reviewer: ```suggestion
To set up, create, build, publish, and deploy agent networks in a CI/CD environment, use the Anypoint CLI plugin. For more information, see xref:af-build-agent-networks-in-a-ci-cd-environment.adoc[].
```
